diff -ruNp libemergent-2.27.4.20608.orig/opt/EVT/myricom_mva/sbin/rebuild.sh esdk-2.27.4.20608/opt/EVT/myricom_mva/sbin/rebuild.sh
--- libemergent-2.27.4.20608.orig/opt/EVT/myricom_mva/sbin/rebuild.sh	2021-03-09 23:28:42.000000000 +0100
+++ libemergent-2.27.4.20608/opt/EVT/myricom_mva/sbin/rebuild.sh	2021-11-25 23:01:23.097714870 +0100
@@ -17,13 +17,6 @@ else
     kver=`uname -r`
 fi
 
-if test ! -d "/lib/modules/$kver"; then
-    echo "'/lib/modules/$kver/build' doesn't exits. Check your kernel version"
-    echo "and/or that you have installed the correct packages to enable compiling"
-    echo "of kernel modules."
-    usage
-fi
-
 if test -n "$1"; then
     prefix=$1
 elif test -n "$MVA_PREFIX"; then
@@ -68,8 +61,8 @@ echo "Building myri_mva.ko for `uname -r
 (
   set -x
   set -e
-  $prefix/src/driver/linux/mal_check_headers.sh /lib/modules/`uname -r`/source \
-  	/lib/modules/`uname -r`/build $prefix/src/driver/linux/mal_checks.h
+  $prefix/src/driver/linux/mal_check_headers.sh $2 \
+  	$3 $prefix/src/driver/linux/mal_checks.h
   if test -d "$prefix/src/mva"; then
   	mva_srcs=`find $prefix/src/mva -maxdepth 1 -type f -name '*.c' | sed -e 's,.*/,,' -e 's/\.c$/.o/'`
   fi
@@ -95,8 +88,8 @@ EXTRA_CFLAGS += -I$prefix/src/driver/zli
 myri_mva-objs += adler32.o crc32.o infblock.o infcodes.o inffast.o inflate.o inftrees.o infutil.o trees.o uncompr.o zutil.o
 endif
 EOF
-  make -C /lib/modules/`uname -r`/build M=$prefix/src/driver/linux/kbuild clean
-  make -C /lib/modules/`uname -r`/build M=$prefix/src/driver/linux/kbuild
+  make -C $3 M=$prefix/src/driver/linux/kbuild clean
+  make -C $3 M=$prefix/src/driver/linux/kbuild
   cp $prefix/src/driver/linux/kbuild/myri_mva.ko $prefix/sbin/
   sed -e "s,MYRI_MODULE_DIR=/opt/mva,MYRI_MODULE_DIR=$prefix," < $prefix/sbin/myri_start_stop.orig > $prefix/sbin/myri_start_stop
   chmod +x $prefix/sbin/myri_start_stop
